apps/snitch_core/test/data/model/product_test.exs

```diff
@@ -152,21 +155,15 @@ defmodule Snitch.Data.Model.ProductTest do
     end
   end
 
-  describe "upi generation for a product" do
-    setup do
-      product = insert(:product)
-      [product: product]
-    end
-
-    test "if no matching upi exists", %{product: product, valid_params: vp} do
-      {:ok, %ProductSchema{} = new_product} = Product.create(vp)
-      refute product.upi == new_product.upi
-    end
-
-    test "if a product with generated upi exists", %{product: product} do
-      assert_raise Ecto.ConstraintError, fn ->
-        insert(:product, %{upi: product.upi})
-      end
+  test "upi generation for a product", %{valid_params: vp} do
+    with_mock GenNanoid, gen_nano_id: fn -> NanoidMock.gen_nano_id() end do
+      NanoidMock.start_link(0)
+      {:ok, product1} = Product.create(vp)
+      vp = %{vp | name: "latest test product"}
+      {:ok, product2} = Product.create(vp)
+      assert product1.upi == "A0C"
+      refute product1.upi == product2.upi
+      NanoidMock.stop()
```

Comment:
@jyotigautam Please group the instructions and leave spaces in between.

Reply:
Done the formatting.